src/Core/GitEventsMonitor.cs

```diff
@@ -0,0 +1,77 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+
+namespace SonarLint.VisualStudio.Core
+{
+    /// <summary>
+    /// Raises events for changes to the git repo
+    /// </summary>
+    public interface IGitEvents
+    {
+        /// <summary>
+        /// Raised when the current head changes
+        /// </summary>
+        event EventHandler HeadChanged;
+    }
+
+    public class GitEventsMonitor : IGitEvents
```

Comment:
This class is expected to be short-lived i.e. a new one created each time a solution is opened.
Since it isn't expected to live for the full lifetime of VS, it should clean up after itself

i.e. the file system watcher is disposable, so this class should be disposable and clean it up correctly (might as well make this class `sealed` as part of this, since it will make the disposal code simpler).